websites/C/Coursera/presence.ts

```diff
@@ -0,0 +1,341 @@
+					},
+				],
+			},
+			{
+				path: /^\/learn\/.+\/exam\/*/,
+				smallImageKey: () => Assets.Writing,
+				smallImageText: () => "Learning",
+				state: () => {
+					return "Solving an Exam";
+				},
+				details: () => {
+					return href.split("/")[4].split("-").join(" ");
+				},
+				buttons: () => [
+					{
+						label: "View Course",
+						url: (() => {
+							return href.split("/").slice(0, 5).join("/");
+						})(),
+					},
+				],
+			},
+			{
+				path: /^\/learn\/.+\/ungraded*\/*/,
+				smallImageKey: () => Assets.Writing,
+				smallImageText: () => "Learning",
+				state: () => {
+					return `Subject: ${document.title.substring(
+						0,
+						document.title.lastIndexOf(" ") - 1
+					)}`;
+				},
+				details: () => {
+					return href.split("/")[4].split("-").join(" ");
+				},
+				buttons: () => [
+					{
+						label: "View Course",
+						url: (() => {
+							return href.split("/").slice(0, 5).join("/");
+						})(),
+					},
+				],
+			},
+
+			{
+				path: /^\/learn\/.*\/?$/,
+				smallImageKey: () => Assets.Viewing,
+				smallImageText: () => "Viewing",
+				details: () => `
+				Viewing a Course`,
+				state: () => {
+					return document.querySelector("[data-e2e='hero-title']").textContent;
+				},
+				buttons: () => [{ label: "Course Link", url: href }],
+			},
+			{
+				path: /^\/specializations\/.+$/,
+				smallImageKey: () => Assets.Viewing,
+				smallImageText: () => "Viewing",
+				details: () => `
+				Viewing a Course`,
+				state: () => {
+					return document.querySelector("[data-e2e='hero-title']").textContent;
+				},
+				buttons: () => [{ label: "Course Link", url: href }],
+			},
+			{
+				path: /^\/search*/,
+				smallImageKey: () => Assets.Search,
+				smallImageText: () => "Searching",
+				details: () => {
+					return `searching for "${new URL(href).searchParams.get("query")}"`;
+				},
+
+				state: () => {
+					return `Found ${
+						document
+							.querySelector("[data-e2e='NumberOfResultsSection']")
+							.textContent.split(" ")[0]
+					} results`;
+				},
+				buttons: () => [{ label: "Results", url: href }],
+			},
+			{
+				path: /^\/+.*$/,
+				details: () => {
+					const currentUrl = href;
+					if (currentUrl.includes("my-learning")) return "My Courses";
+					else
+						return currentUrl.split("/")[3].toUpperCase().split("-").join(" ");
+				},
+				smallImageKey: () => Assets.Reading,
+				smallImageText: () => "Browsing",
+			},
+		];
+
+		return routes.find(route => route.path.test(path));
+	};
+
+presence.on("UpdateData", async () => {
+	const [showTimestamp, showButtons] = await Promise.all([
+			presence.getSetting<boolean>(Settings.TIMESTAMP),
+			presence.getSetting<boolean>(Settings.BUTTONS),
+		]),
+		presenceData: PresenceData = {
+			largeImageKey: Assets.Logo,
+		};
+
+	if (showTimestamp) presenceData.startTimestamp = startTimestamp;
+
+	const route = router({
+		presenceData,
+		path: location.href.replace(`https://${location.hostname}`, ""),
```

Comment:
Use `document.location` rather than `window.location` for consistency.
```suggestion
		path: href.replace(`https://${location.hostname}`, ""),
```
```suggestion
		path: document.location.href.replace(`https://${location.hostname}`, ""),
```